nixos/modules/services/networking/xl2tpd.nix

```diff
@@ -124,13 +153,31 @@ with lib;
         chown root.root l2tp-secrets
         chmod 600 l2tp-secrets
 
+        cat > ppp/ip-up << EOF
+        #!/bin/sh
+        export PATH=${path}
+        ${cfg.up}
+        EOF
```

Comment:
This looks like a dangerous way to generate the ip-up/down scripts. What if `cfg.up` contains the string `EOF`? Seems better to use `pkgs.writeScript`.